stdlib/public/core/Policy.swift

```diff
@@ -15,8 +15,26 @@
 //===----------------------------------------------------------------------===//
 // Standardized uninhabited type
 //===----------------------------------------------------------------------===//
-/// The return type of functions that do not return normally, that is, a type
-/// with no values.
+/// The return type of functions that cannot return a value.
+///
+/// `Never` is a type with no values and cannot be constructed, allowing you
+/// to represent a state in your program that is impossible to reach during
+/// its execution.
+///
+///     // An infinite loop will never terminate.
+///     func forever() -> Never {
+///         while true {
+///             print("I will print forever and will not return.")
+///         }
+///     }
+///
+///     // The `.failure` case cannot be reached because it cannot be 
+///     // constructed; this makes the following switch exhaustive.
+///     let favouriteNumber: Result<Int, Never> = .success(42)
+///     switch favouriteNumber {
+///     case .success(let value):
+///         print("My favourite number is", value)
```

Comment:
US English spelling?

```suggestion
///     let favoriteNumber: Result<Int, Never> = .success(42)
///     switch favoriteNumber {
///     case .success(let value):
///         print("My favorite number is", value)
```